src/Management/src/EndpointCore/SpringBootAdminClient/BootAdminClientOptions.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Steeltoe.Management.Endpoint.SpringBootAdminClient
+{
+    public class BootAdminClientOptions
+    {
+        private const string PREFIX = "spring:boot:admin:client";
+        private const string APPLICATION_NAME_PREFIX = "spring:application:name";
+        private const string APPLICATION_NAME_FALLBACK = "ApplicationName";
+        private const string URLS = "URLS";
+
+        public string Url { get; set; }
+
+        public string ApplicationName { get; set; }
+
+        public string BasePath { get; set; }
+
+        public BootAdminClientOptions(IConfiguration config)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+
+            var section = config.GetSection(PREFIX);
+            if (section != null)
+            {
+                section.Bind(this);
+            }
+
+            BasePath ??= GetBasePath(config);
+            ApplicationName ??= config.GetValue<string>(APPLICATION_NAME_PREFIX);
```

Comment:
Ideally all of this type of logic should now live in `Steeltoe.Common.ApplicationInstanceInfo`, is that move workable?